code/code/misc/parse.cc

```diff
@@ -2081,16 +2083,8 @@ void argument_interpreter(sstring argument, sstring &first_arg, sstring &second_
 
 bool is_number(const sstring &str)
 {
-  int look_at;
-
-  if (str.empty())
-    return (0);
-
-  for (look_at = 0; *(str.c_str() + look_at) != '\0'; look_at++) {
-    if ((*(str.c_str() + look_at) < '0') || (*(str.c_str() + look_at) > '9'))
-      return (0);
-  }
-  return (1);
+    boost::smatch what;  //Not really used here, since we're just doing go/no-go
+    return boost::regex_match(str, what, boost::regex("^[\\+-]?\\d+(\\.\\d+)?$"));
```

Comment:
I'd've used `R"(^[\+-]?\d+(\.\d+)?$)"` -- less confusing backslashes this way.

Reply:
I didn't even see this until just now. I did exactly that a little bit earlier.